src/Shared/NodePipeBase.cs

```diff
@@ -226,7 +226,7 @@ private int Read(byte[] buffer, int bytesToRead)
         }
 
 #if !TASKHOST
-        private async ValueTask<int> ReadAsync(byte[] buffer, int bytesToRead, CancellationToken cancellationToken)
+        private async Task<int> ReadAsync(byte[] buffer, int bytesToRead, CancellationToken cancellationToken)
```

Comment:
why? not sure if it applies here but wouldn't it potentially cause extra allocations? https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/#valuetask-<tresult>-and-synchronous-completion

Reply:
I mentioned here (https://github.com/dotnet/msbuild/pull/11383#issuecomment-2759731576), but I hit compile issues outside of `Microsoft.Build` since I don't believe we actually pull in `System.Threading.Tasks.Extensions` anywhere, and my attempts to fix it result in additional pre-builts. Happy to switch back to ValueTask if there's an easy solution here.

Reply:
I think adding `Microsoft.Bcl.AsyncInterfaces.9.0.*` to [sourcebuild baseline](https://github.com/dotnet/msbuild/blob/main/eng/SourceBuildPrebuiltBaseline.xml) should be fine. @MichaelSimons do you see any issue with that?

Reply:
If you are not declaring an explicit dependency on Microsoft.Bcl.AsyncInterfaces and it is coming in from System.Threading.Channels then it would be correct that it can be added to the prebuilt baseline.

FYI, please utilize the @dotnet/source-build team for requests like this as it is monitored by more folks.

Reply:
Awesome - added the prebuilt, tested on the other PR w/ the channels reference, and it worked. Restored the `ValueTask` change.